manifests/install.pp

```diff
@@ -23,9 +23,21 @@
     ensure => $ensure
   }
 
-  package { ['zookeeperd']: #init.d scripts for zookeeper
-    ensure  => $ensure,
-    require => Package['zookeeper']
+
+  if $::lsbdistcodename == 'xenial' {
+      file { '/usr/lib/systemd/system/zookeeper.service':
+          mode    => '0664',
+          source  => 'puppet:///modules/zookeeper/zookeeper.service',
+          require => Package['zookeeper'],
+      } ~> exec { 'systemctl daemon-reload':
+            refreshonly => true,
+            path        => $::path,
+      }
+  } else {
+    package { ['zookeeperd']: #init.d scripts for zookeeper
```

Comment:
So to confirm, on xenail we 'dont' need this package at all? On xenail does *it* come with a systemd service file?

Reply:
This surprises me. Are we trying to use trusty versions of these packages on xenial hosts?

Reply:
We are not using trusty version of package on xenial hosts. (We don't maintain the zookeeper package ourselves, we use whatever is the one for that distro, which is https://packages.ubuntu.com/xenial/zookeeper (3.4.8-1) for xenial).


And, neither zookeeperd nor any package in xenial relating to zookeeper, ships with a unit file for systemd:  https://packages.ubuntu.com/search?keywords=zookeeper&searchon=all&suite=xenial&section=all